lock/src/main/java/com/auth0/android/lock/provider/OAuth2WebAuthProvider.java

```diff
@@ -137,31 +139,44 @@ private void startAuthorization(Activity activity, Uri authorizeUri, String conn
     @Override
     public boolean authorize(Activity activity, @NonNull AuthorizeResult data) {
         if (!data.isValid(OAUTH2_REQUEST_CODE)) {
+            Log.w(TAG, "Invalid request.");
```

Comment:
At this point, I can only know if it succeeded or failed, not the cause. Maybe the logging belongs to `AuthorizeResult.isValid()`